### PR TITLE
Fix #2539: Fix java.util.ConcurrentHashMap#putIfAbsent.

### LINKED ISSUE #2539: java.util.ConcurrentHashMap#putIfAbsent is wrong
The documentation of `putIfAbsent` specifies that it returns `null` if the key doesn't exist, while the implementation uses `getOrElseUpdate` which specifies that either the existing or the new value is returned.

Other classes might also be affected.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala [hunk 1]
```diff
@@ -50,6 +50,13 @@ class ConcurrentHashMapTest extends MapTest {
     assertEquals("one", chm.get("ONE"))
   }
 
+  @Test def putIfAbcentShould_issue_2539(): Unit = {
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala]: That test name looks weird. Consider `testPutIfAbsent_issue_2539`.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala [hunk 2]
```diff
@@ -50,6 +50,13 @@ class ConcurrentHashMapTest extends MapTest {
...
+    val chm = factory.empty[String, String]
+    assertNull(chm.putIfAbsent("abc", "def"))
+    assertEquals("def", chm.putIfAbsent("abc", "def"))
+    assertEquals("def", chm.putIfAbsent("abc", "ghi"))
+    assertEquals("def", chm.putIfAbsent("abc", "jkl"))
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala]: Maybe worth testing `putIfAbsent` with something else than `"abc"` and verify that it returns `null` again, for a different key.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala [hunk 3]
```diff
@@ -50,6 +50,13 @@ class ConcurrentHashMapTest extends MapTest {
     assertEquals("one", chm.get("ONE"))
   }
 
+  @Test def putIfAbcentShould_issue_2539(): Unit = {
+    val chm = factory.empty[String, String]
+    assertNull(chm.putIfAbsent("abc", "def"))
+    assertEquals("def", chm.putIfAbsent("abc", "def"))
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala]: Maybe test `assertEquals("def", chm.get("abc"))` before that line.

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**